### DISCUSSION TITLE
Is it possible to rotate a PDF by a certain angle such as 30 degress?

### DISCUSSION THREAD

**Discussion by ilovefreesw** [py-pdf/pypdf#2267] [General]
I have some skewed PDFs, by around 30 degrees. So, I need to rate them. Can PyDF2 do that?

**Comment by stefan6419846** [comment]
PyPDF2 is deprecated, please migrate to `pypdf`.

For the remaining stuff, see https://pypdf.readthedocs.io/en/latest/user/cropping-and-transforming.html#page-rotation: You can use `page.rotate(90)`, but only with angles which are a multiple of 90 degrees as per the docs. Using `Transformation.rotate` and `page.add_transformation(matrix)`, you might achieve arbitrary rotations, but it might require further work to fix the page itself.

**Reply by ilovefreesw** [reply]
Thanks
